### PR TITLE
only start appinsights threads with configured app insights instances

### PR DESCRIPTION
**Pull request by bmc-msft**
## Summary of the Pull Request

Only launches app-insights threads if app-insights are configured

## PR Checklist
* [ ] Applies to work item: #xxx
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [x] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

This only launches app-insights instances if they are configured

## Validation Steps Performed

Run standard unit tests

### CODE REVIEW

File: src/agent/onefuzz-supervisor/src/main.rs [hunk 1]
```diff
@@ -145,7 +145,7 @@ async fn run_agent(config: StaticConfig) -> Result<()> {
         Box::new(worker::WorkerRunner),
     );
 
-    info!("running supervisor agent");
+    info!("running agent");
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/main.rs:148]: Why be less specific?

File: src/agent/onefuzz-supervisor/src/config.rs [hunk 2]
```diff
@@ -111,13 +111,12 @@ impl Registration {
...
-            .append_pair("pool_name", &config.pool_name);
+            .append_pair("pool_name", &config.pool_name)
+            .append_pair("version", env!("ONEFUZZ_VERSION"));
 
         if managed {
             let scaleset = onefuzz::machine_id::get_scaleset_name().await?;
-            url.query_pairs_mut()
-                .append_pair("scaleset_id", &scaleset)
-                .append_pair("version", env!("ONEFUZZ_VERSION"));
+            url.query_pairs_mut().append_pair("scaleset_id", &scaleset);
```
Comment (ranweiler) [inline src/agent/onefuzz-supervisor/src/config.rs:119]: Is this change related?

File: src/agent/onefuzz/Cargo.toml [hunk 3]
```diff
@@ -20,6 +20,7 @@ notify = "4.0"
 regex = "1.3"
 reqwest = { version = "0.10", features = ["json", "stream"] }
 sha2 = "0.9"
+url = { version = "2.1.1", features = ["serde"] }
```
Comment (ranweiler) [inline src/agent/onefuzz/Cargo.toml:23]: Is this used in this PR?

**Review by ranweiler — APPROVED** [review]

File: src/agent/onefuzz-supervisor/src/config.rs (hunk 2 above)
Comment (bmc-msft) [reply src/agent/onefuzz-supervisor/src/config.rs:119]: No.   This PR was split off from a larger one.  That said, this is needed.  Going to leave it here.

File: src/agent/onefuzz/Cargo.toml (hunk 3 above)
Comment (bmc-msft) [reply src/agent/onefuzz/Cargo.toml:23]: No.  Incidental commit from a PR than I'd split it off.  Since the other is PR is coming, just going to leave this here.

File: src/agent/onefuzz-supervisor/src/main.rs (hunk 1 above)
Comment (bmc-msft) [reply src/agent/onefuzz-supervisor/src/main.rs:148]: This is more correct.  it's launching onefuzz-agent

Comment (ranweiler) [reply src/agent/onefuzz-supervisor/src/main.rs:148]: @bmc-msft, FWIW, it really is invoking the _supervisor_ agent. The worker agent (`onefuzz-agent`) is invoked in [`WorkerRunner::run()`](https://github.com/bmc-msft/onefuzz/blob/a998f860dca4cb21740b60a56eb094bfb939b3da/src/agent/onefuzz-supervisor/src/worker.rs#L216).

Also, there is now a "supervisor agent finished" log message a few lines below that no longer matches this one.
